crates/vm/levm/src/opcode_handlers/push.rs

```
@@ -64,12 +64,13 @@ fn read_bytcode_slice(current_call_frame: &CallFrame, n_bytes: usize) -> Result<
 
     Ok(current_call_frame
         .bytecode
-        .get(pc_offset..)
-        .unwrap_or_default()
-        .iter()
-        .take(n_bytes)
-        .cloned()
-        .collect())
+        .get(
+            pc_offset
+                ..pc_offset
+                    .checked_add(n_bytes)
+                    .ok_or(InternalError::SlicingError)?,
```

Are we sure this should return an error instead of all zeroes (like [CODECOPY](https://www.evm.codes/?fork=cancun#39)) when reading outside bounds?

---

The amount of tests that pass remains the same but returning out of bounds error seems more correct so I changed it in [return outofbounds instead of internal error](https://github.com/lambdaclass/ethrex/pull/1598/commits/88c486584a9dbe74e45ac8ac130019631fe35a17)

---

I think there is an edge case in which we can end the bytecode with a PUSHN and a value that is less than N bytes. This can only happen if we are at the end of the bytecode.
So in this case the `unwrap_or_default()` wouldn't produce the right behavior because it would push zeros to the stack instead of the value but it won't matter that much because a `PUSH` at the end of the bytecode won't impact in the resulting state.
[For example](https://www.evm.codes/playground?fork=cancun&unit=Wei&codeType=Bytecode&code='7f04'_)

I remember that @maximopalopoli handled that in the past

---

And about the out of bounds read that @MegaRedHand said. In this case it won't ever happen that `pc_offset + n_bytes` overflows because `pc_offset` is limited to the max bytecode size (24576 bytes) and n_bytes's max value is 32. If that happens it means we are in trouble I guess, so an `InternalError` doesn't sound bad to me.